include/ht_shadervariable.h

```diff
@@ -73,21 +73,21 @@ namespace Hatchit {
         template<>
         inline void HT_API ShaderVariableTemplate<int>::SetData(int data)
         {
+            m_data = new BYTE[sizeof(int)];
```

Comment:
Why not instead of allocating and re-interpreting a byte array as the specified type, we just call new int, new bool, new double, etc. ?